### PR TITLE
handle task config NPEs

### PR DESCRIPTION
**Pull request by iamnoah**

### CODE REVIEW

File: sink/src/main/java/com/spredfast/kafka/connect/s3/sink/S3SinkTask.java [hunk 1]
```diff
@@ -154,21 +150,23 @@ public void flush(Map<TopicPartition, OffsetAndMetadata> offsets) throws Connect
...
 
 	private String name() {
-		return config.get("name");
+		return configGet("name").orElseThrow(() -> new IllegalWorkerStateException("Tasks always have names"));
 	}
 
 	public void finishWriter(BlockGZIPFileWriter writer, TopicPartition tp) throws IOException {
-		writers.get(tp).finish(tp.topic(), tp.partition());
+		writerGet(tp).ifPresent(w -> w.finish(tp.topic(), tp.partition()));
 		writers.remove(tp);
```
Comment (cfstout) [inline sink/src/main/java/com/spredfast/kafka/connect/s3/sink/S3SinkTask.java]: Will the remove call fail if it's not present?

**Comment by epollan** [comment]
👍

File: source/src/main/java/com/spredfast/kafka/connect/s3/source/S3SourceTask.java [hunk 2]
```diff
@@ -75,10 +75,10 @@ private void readFromStoredOffsets() {
 	}
 
 	private void tryReadFromStoredOffsets() throws UnsupportedEncodingException {
-		String bucket = this.taskConfig.get("s3.bucket");
-		String prefix = this.taskConfig.get("s3.prefix");
+		String bucket = configGet("s3.bucket").orElseThrow(() -> new ConnectException("No bucket configured!"));
+		String prefix = configGet("s3.prefix").orElse(null);
```
Comment (cfstout) [inline source/src/main/java/com/spredfast/kafka/connect/s3/source/S3SourceTask.java]: Above this is defaulted to `.orElse("")` why is this different?

File: sink/src/main/java/com/spredfast/kafka/connect/s3/sink/S3SinkTask.java [hunk 3]
```diff
@@ -57,28 +57,20 @@ public String version() {
...
-		String chunkThreshold = config.get("compressed_block_size");
-		if (chunkThreshold != null) {
-			try {
-				this.GZIPChunkThreshold = Long.parseLong(chunkThreshold);
-			} catch (NumberFormatException nfe) {
-				// keep default
-			}
-		}
+
+		configGet("compressed_block_size").map(Long::parseLong).ifPresent(chunkThreshold ->
```
Comment (AlexMoffat) [inline sink/src/main/java/com/spredfast/kafka/connect/s3/sink/S3SinkTask.java]: What about the NumberFormatException?

File: source/src/main/java/com/spredfast/kafka/connect/s3/source/S3SourceTask.java [hunk 4]
```diff
@@ -93,39 +93,43 @@ private void tryReadFromStoredOffsets() throws UnsupportedEncodingException {
...
 			.orElse(10_000L);
-		errorBackoff = Optional.ofNullable(taskConfig.get("s3.error.backoff"))
+		errorBackoff = configGet("s3.error.backoff")
 			.map(Long::parseLong)
 			.orElse(1000L);
 
 		AmazonS3 client = S3.s3client(taskConfig);
 
-		Set<String> topics = Optional.ofNullable(taskConfig.get("topics"))
+		Set<String> topics = Optional.ofNullable(configGet("topics"))
```
Comment (cfstout) [inline source/src/main/java/com/spredfast/kafka/connect/s3/source/S3SourceTask.java]: Double wrapping here? Looks like you'll have an `Optional<Optional<String>>`

File: source/src/main/java/com/spredfast/kafka/connect/s3/source/S3SourceTask.java (hunk 2 above)
Comment (AlexMoffat) [inline source/src/main/java/com/spredfast/kafka/connect/s3/source/S3SourceTask.java]: The other default for prefix earlier on was "" so is null right here?

File: source/src/main/java/com/spredfast/kafka/connect/s3/source/S3SourceTask.java (hunk 4 above)
Comment (iamnoah) [inline source/src/main/java/com/spredfast/kafka/connect/s3/source/S3SourceTask.java]: Good catch.

File: source/src/main/java/com/spredfast/kafka/connect/s3/source/S3SourceTask.java (hunk 2 above)
Comment (iamnoah) [inline source/src/main/java/com/spredfast/kafka/connect/s3/source/S3SourceTask.java]: I thought the tasks had different slightly different null semantics, but I don't think they do. "" seems like a better default.

File: sink/src/main/java/com/spredfast/kafka/connect/s3/sink/S3SinkTask.java (hunk 1 above)
Comment (iamnoah) [inline sink/src/main/java/com/spredfast/kafka/connect/s3/sink/S3SinkTask.java]: Nope, just a Map.

File: sink/src/main/java/com/spredfast/kafka/connect/s3/sink/S3SinkTask.java (hunk 3 above)
Comment (iamnoah) [inline sink/src/main/java/com/spredfast/kafka/connect/s3/sink/S3SinkTask.java]: It will bubble up, kill the task, and be reported.

**Pull request merged**